File: JavaScript/4-类和对象.js
class BBB{

}

class AAA extends BBB{
    construct(val){
        this.a1 = val;
    }
    getA1(){
        return this.a1;
    }
    setA1(val){
        this.a1 = val;
    }
    static staticA1(){
        return "static method";
    }
}

let aaa = new AAA(5);
console.log(aaa.getA1());
aaa.setA1(2);
console.log(aaa.getA1());

File: C#/4-类和对象.cs
// 接口
interface BBB{
    void bbMethod();
}

// 类定义
class AAA : BBB
{
    private double a1;
    public void setA1( double a1 ){
        a1 = val;
    };
    public double getA1( void ){
        return a1;
    };
    public AAA(val){  // 构造函数
        Console.writeLine("construct");
        a1 = val;
    };
    public void bbMethod(){  // 实现接口
        Console.writeLine("implement bbMehod");
    }
    static void Main(string[] args) // 静态方法
    {
         AAA aaa = new AAA(10.0);
         Console.WriteLine(aaa.getLength());
         aaa.setLength(6.0);
         Console.WriteLine(aaa.getLength());
         Console.ReadKey();
    }
};

// 对象定义
int main(){
    double temp = 0.0;
    AAA aaa;
    aaa.setA1(5.0);
    temp = aaa.getA1();
    cout<< temp;
}